Get rid of node-canvas dependency

Some team members get lots of troubles while they install fabricjs because of the issues of installing node-canvas. I also got some troubles but I'm happy to be an macos user and it wasn't hard to fix it. Windows users aren't so happy with that :D

We use fabricjs only for browser-side development and we don't need any features for nodejs. Is that possible to remove "canvas" from "optionalDependencies", add it to "devDependencies" and leave a note at README that if somebody wants to use node-canvas then he/she needs to install it manually? 

A good bonus would be also to get rid of "jsdom" dependency because it just gives a package more weight.

In our project, we are using webpack, react and serverside rendering, and after we had upgraded fabric to 3.1, we started to get problems with node-canvas. After deploying to the server, on server rendering we started to get the error 'Cannot find module '../build/Release/canvas.node' at module.js'. The strange thing is that there is no code connected with node-canvas in the bundled file. Another strange one is that the error disappears after several reloads of the page. After deploying, js file is being bundled only once, and after reloading the page it could not be changed, but suddenly the error had gone, and I cannot get why. Tried to use webpack config with configuring externals, that was mentioned earlier and it does not help. Helps only downgrading to 2.7 as it uses an earlier version of node-canvas. My guess is that in node-canvas has changed the way of installing in version 2.0. But why it is trying to be installed if it is optional dependency and was not required anywhere, still to be a question?

--- a/publish.js
+++ b/publish.js
@@ -1,0 +1,31 @@
+var cp = require('child_process');
+var path = require('path');
+var fs = require('fs');
+
+// eslint-disable-next-line no-undef
+var pkgPath = path.resolve(__dirname, './package.json');
+var pkgText = fs.readFileSync(pkgPath); // get original pkg text to restore it later
+var pkgObject = JSON.parse(pkgText); // parsed pkg to override its fields
+var args = process.argv.slice(2).join(' '); // args will be passed to npm publish (like --dry-run)
+
+// override package.json with updated fields
+fs.writeFileSync(
+  pkgPath,
+  JSON.stringify(Object.assign(pkgObject, {
+    optionalDependencies: {},
+    version: pkgObject.version + '-browser',
+  }), null, '\t')
+);
+
+// publish -browser version
+cp.execSync('npm publish ' + args);
+
+console.log('Browser package is published');
+
+// restore the original package.json contents
+fs.writeFileSync(pkgPath, pkgText);
+
+// publish the main version (the package is published "above" the -browser version)
+cp.execSync('npm publish ' + args);
+
+console.log('Main package is published');

